public/app/features/dashboard/containers/DashboardPage.tsx

```diff
@@ -90,8 +100,13 @@ export class DashboardPage extends PureComponent<Props, State> {
   }
 
   componentWillUnmount() {
-    if (this.props.dashboard) {
-      this.props.cleanUpDashboard();
+    if (this.props.urlUid !== this.state.prevUrlUid && this.state.prevUrlUid !== null) {
+      this.props.cleanUpDashboardState(this.state.prevUrlUid);
+      this.setPanelFullscreenClass(false);
+    }
+
+    if (this.props.initPhase === DashboardInitPhase.Completed && this.state.prevUrlUid === null) {
+      this.props.cleanUpDashboardState(this.state.prevUrlUid);
       this.setPanelFullscreenClass(false);
     }
```

Comment:
Here is where I get stuck solving cleaning up state after unmount. It feels like I'm close to a solution, but I know too little about the different navigation states and bridgeSrv to figure out a solution that will always work. Most of the times I get unmount for the previous dashboard uid but sometimes I get the currently mounted uid.

Reply:
I solved this, when we receive componentWillUnmount the "new" dashboards uId is in location state. So all that we need todo is to clear any previous completed dashboard. Created a Thunk called `cleanUpUnCleanedDashboardStates` that solves this.